### ISSUE TITLE
Support for schema evolution in create table statements

### ISSUE THREAD

**Issue by epilif1017a** [delta-io/delta#963]
When we drop + create a delta lake table, if the schema defined in the create table statement does not match the schema stored in the distributed file system we get the following exception:

```
Error in SQL statement: AnalysisException: The specified schema does not match the existing schema at dbfs:/tmp/my.db/test1.

== Specified ==
root
-- a: string (nullable = true)
-- b: map (nullable = true)
    |-- key: string
    |-- value: integer (valueContainsNull = true)
-- c: string (nullable = true)
-- d: struct (nullable = true)
    |-- a: string (nullable = true)
    |-- b: integer (nullable = true)


== Existing ==
root
-- a: string (nullable = true)
-- b: string (nullable = true)
-- c: string (nullable = true)
-- d: struct (nullable = true)
    |-- a: string (nullable = true)


== Differences==
- Specified type for b is different from existing schema:
  Specified: map
  Existing:  string
- Specified schema has additional field(s): d.b

If your intention is to keep the existing schema, you can omit the
schema from the create table command. Otherwise please ensure that
the schema matches.
```

**Intended behaviour:** there should be an option similar to spark.databricks.delta.schema.autoMerge.enabled (or even make this option work for create tables) that allows delta lake to evolve the schema while creating a new version of the table through a create table command, otherwise users need to fire several alter table commands to make the schema match.

**Comment by epilif1017a** [comment]
@zsxwing thanks for looking into it, I think this would be a great addition, because in a mostly sql based ddl cicd process it's a bit hard to introduce the mechanics proposed above, for example. So if we could simply put a flag in the spark session, to allow create table schema merge, we would be leaving the dream :) 

Thanks for driving the project forward!

**Comment by zsxwing** [comment]
@epilif1017a Yep. Adding SQL support makes sense given that this is already supported by DataFrame APIs.

**Comment by foss-contributor** [comment]
@epilif1017a, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
